t/plugin/elasticsearch-logging.t

```diff
@@ -0,0 +1,299 @@
+                }
+            }
+
+            local plugin = require("apisix.plugins.elasticsearch-logging")
+            for i = 1, #configs do
+                ok, err = plugin.check_schema(configs[i])
+                if err then
+                    ngx.say(err)
+                else
+                    ngx.say("passed")
+                end
+            end
+        }
+    }
+--- response_body_like
+passed
+passed
+property "endpoint" validation failed: property "uri" is required
+property "endpoint" validation failed: property "index" is required
+property "endpoint" validation failed: property "uri" validation failed.*
+
+
+
+=== TEST 2: set route
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1', ngx.HTTP_PUT, {
+                uri = "/hello",
+                upstream = {
+                    type = "roundrobin",
+                    nodes = {
+                        ["127.0.0.1:1980"] = 1
+                    }
+                },
+                plugins = {
+                    ["elasticsearch-logging"] = {
+                        endpoint = {
+                            uri = "http://127.0.0.1:9200",
+                            index = "services"
+                        },
+                        batch_max_size = 1,
+                        inactive_timeout = 1
+                    }
+                }
+            })
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: test route (success write)
+--- request
+GET /hello
+--- wait: 2
+--- response_body
+hello world
+
+
+
+=== TEST 4: set route (auth)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1', ngx.HTTP_PUT, {
+                uri = "/hello",
+                upstream = {
+                    type = "roundrobin",
+                    nodes = {
+                        ["127.0.0.1:1980"] = 1
+                    }
+                },
+                plugins = {
+                    ["elasticsearch-logging"] = {
+                        endpoint = {
+                            uri = "http://127.0.0.1:9201",
+                            index = "services",
+                            username = "elastic",
+                            password = "123456"
+                        },
+                        batch_max_size = 1,
+                        inactive_timeout = 1
+                    }
+                }
+            })
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 5: test route (auth success)
+--- request
+GET /hello
+--- wait: 2
+--- response_body
```

Comment:
We should check the data sent to the elasticsearch, via injecting like: https://github.com/apache/apisix/pull/7593#issuecomment-1210208755

Reply:
> We should check the data sent to the elasticsearch, via injecting like: [#7593 (comment)](https://github.com/apache/apisix/pull/7593#issuecomment-1210208755)

Does you mean I should hook the function `httpc:request_uri` and mock it with my own mock_request_uri, and check the request body with the mock_request_uri?
If so, I am not quite understand the nessary to check request body.

Reply:
> > We should check the data sent to the elasticsearch, via injecting like: [#7593 (comment)](https://github.com/apache/apisix/pull/7593#issuecomment-1210208755)
> 
> Does you mean I should hook the function `httpc:request_uri` and mock it with my own mock_request_uri, and check the request body with the mock_request_uri? If so, I am not quite understand the nessary to check request body.

@spacewander 
Please pay attention to this question when you are free, Thank you.

Reply:
Sorry for missing this. Yes, you are right.